plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSink.java

```diff
@@ -361,7 +364,9 @@ private void closeWriter(HiveWriter writer)
             HiveWriter writer = writers.get(writerIndex);
             if (writer != null) {
                 // if current file not too big continue with the current writer
-                if (bucketFunction != null || writer.getWrittenBytes() <= targetMaxFileSize.orElse(Long.MAX_VALUE)) {
+                // for transactional tables we don't want to split output files because there is an implicit bucketing with
+                // 1 bucket so there should be only one output file bucket_00000
```

Comment:
```
// for transactional tables we don't want to split output files because there is an explicit or implicit bucketing
// and file names have no random component (e.g. bucket_00000)
```